Tasks/DotnetCLIRestore/restorecommand.ts

```
@@ -0,0 +1,306 @@
+import * as fs from "fs";
+import * as ltx from "ltx";
+import * as tl from 'azure-pipelines-task-lib/task';
+import * as Q from 'q';
+import * as auth from 'packaging-common/nuget/Authentication';
+import { NuGetConfigHelper2 } from 'packaging-common/nuget/NuGetConfigHelper2';
+import * as ngRunner from 'packaging-common/nuget/NuGetToolRunner2';
+import * as path from 'path';
+import { IExecOptions } from 'azure-pipelines-task-lib/toolrunner';
+import * as request from 'request';
+
+export async function run(): Promise<void> {
+    const buildIdentityDisplayName: string = null;
+    const buildIdentityAccount: string = null;
+
+    try {
+        let packageName: string = tl.getInput("packageName");
+        let packageVersion: string = tl.getInput("version") || "";
+
+        if (!packageName)
+        {
+            return;
```

it should fail with proper error if a valid package name is not specified